theano/tensor/slinalg.py

```diff
@@ -413,7 +413,12 @@ def perform(self, node, inputs, outputs):
         U = scipy.linalg.inv(V).T
 
         exp_w = numpy.exp(w)
-        X = numpy.subtract.outer(exp_w, exp_w) / numpy.subtract.outer(w, w)
+        exp_diff = numpy.subtract.outer(exp_w, exp_w)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            X = exp_diff / numpy.subtract.outer(w, w)
+        X[exp_diff == 0] = 1
```

Comment:
To answer my own previous question, similarly fixing the division by zero with a Taylor expansion seems to work for the complete derivative:
```
def my_func(x):
    out = np.divide( (1-np.exp(-x)), x )
    idx = (abs(x) < 1e-6)
    x2idx = np.square(x[idx])
    out[idx] = 1 - x[idx]/2 + x2idx/6
    return out

expA = np.expm(A)
IkeA = np.kron(np.eye(3), expA)
adA = np.kron(np.eye(3), A) - np.kron(A.T, np.eye(3))
funadA = scipy.linalg.funm(adA, my_func )
DexpA = np.dot(IkeA, funadA)
```